python/graphstorm/gconstruct/config_conversion/gsprocessing_converter.py

```diff
@@ -0,0 +1,329 @@
+
+                labels_list.append(label_dict)
+            except KeyError as exc:
+                raise KeyError(
+                    f"A required key was missing from label input {label}"
+                ) from exc
+        return labels_list
+
+    @staticmethod
+    def _convert_feature(feats: list[Mapping[str, Any]]) -> list[dict]:
+        """Convert the feature config
+        Parameters
+        ----------
+        feats: list[Mapping[str, Any]]
+            The feature information in the GSProcessing format
+
+        Returns
+        -------
+        list[dict]
+            The feature information in the GConstruct format
+        """
+        gconstruct_feats_list = []
+        if feats in [[], [{}]]:
+            return []
+        for gsp_feat_dict in feats:
+            gconstruct_feat_dict = {}
+            gconstruct_feat_dict["feature_col"] = gsp_feat_dict["column"]
+            if "name" in gsp_feat_dict:
+                gconstruct_feat_dict["feature_name"] = gsp_feat_dict["name"]
+
+            gconstruct_transformation_dict: dict[str, Any] = {}
+            if "transformation" in gsp_feat_dict:
+                gsp_transformation_dict = gsp_feat_dict["transformation"]
+
+                ### START TRANSFORMATION ALTERNATIVES ###
+                if gsp_transformation_dict["name"] == "numerical":
+                    if gsp_transformation_dict["kwargs"]["normalizer"] == "min-max":
+                        gconstruct_transformation_dict["name"] = "max_min_norm"
+                    elif gsp_transformation_dict["kwargs"]["normalizer"] == "standard":
+                        gconstruct_transformation_dict["name"] = "standard"
+                    elif (
+                        gsp_transformation_dict["kwargs"]["normalizer"] == "rank-gauss"
+                    ):
+                        gconstruct_transformation_dict["name"] = "rank_gauss"
+                        if gsp_transformation_dict.get("kwargs").get("epsilon"):
+                            gconstruct_transformation_dict["epsilon"] = (
+                                gsp_transformation_dict["kwargs"]["epsilon"]
+                            )
+                    else:
+                        raise ValueError(
+                            f"Unexpected numerical transformation "
+                            f"{gsp_transformation_dict['kwargs']['normalizer']}"
+                        )
+                    if "out_dtype" in gsp_transformation_dict["kwargs"]:
+                        gconstruct_transformation_dict["out_dtype"] = (
+                            gsp_transformation_dict["kwargs"]["out_dtype"]
+                        )
+                elif gsp_transformation_dict["name"] == "bucket_numerical":
+                    assert (
+                        "bucket_cnt" in gsp_transformation_dict["kwargs"]
+                    ), "bucket_cnt should be in the gconstruct bucket feature transform field"
```

Comment:
Do we mean gsprocessing bucket feature transform field? We're creating the gconstruct transform. Similarly for other similar errors here.

Reply:
Right. We have similar assertion as well.